helix-core/src/main/java/org/apache/helix/monitoring/mbeans/ClusterStatusMonitor.java

```diff
@@ -365,6 +357,35 @@ public void increaseMessageReceived(List<Message> messages) {
     }
   }
 
+  /**
+   * Update max capacity usage for per instance.
+   * @param maxCapacityUsageMap a map of max capacity usage, {instance: maxCapacityUsage}
+   */
+  public void updateMaxCapacityUsage(Map<String, Double> maxCapacityUsageMap) {
+    for (Map.Entry<String, Double> entry : maxCapacityUsageMap.entrySet()) {
+      getOrCreateInstanceMonitor(entry.getKey()).updateMaxCapacityUsage(entry.getValue());
```

Comment:
Instance monitor creating is controlled by setClusterInstanceStatus. Please don't do such logic anywhere else. Unless we refactor this ClusterStatusMonitor completely, which I don't think is a good idea now.
Just try to read whatever instance monitor exists and update. If not found, ignore with a warning.